apps/web-client/src/lib/db/cr-sqlite/order-reconciliation.ts

```diff
@@ -1,64 +1,81 @@
 import { multiplyString } from "./customers";
-import type { DB, ReconciliationOrder } from "./types";
+import type { DB, ReconciliationOrder, ReconciliationOrderLine } from "./types";
 
 export async function getAllReconciliationOrders(db: DB): Promise<ReconciliationOrder[]> {
 	const result = await db.execO<ReconciliationOrder>(
-		"SELECT id, supplier_order_ids, finalized, customer_order_line_ids, created FROM reconciliation_order ORDER BY id ASC;"
+		"SELECT id, supplier_order_ids, finalized, updatedAt, created FROM reconciliation_order ORDER BY id ASC;"
 	);
 	return result;
 }
 
 export async function getReconciliationOrder(db: DB, id: number): Promise<ReconciliationOrder> {
 	const result = await db.execO<ReconciliationOrder>(
-		"SELECT id, supplier_order_ids, finalized, customer_order_line_ids, created FROM reconciliation_order WHERE id = ?;",
+		`SELECT id, supplier_order_ids, finalized, updatedAt, created
+		FROM reconciliation_order WHERE id = ?;`,
 		[id]
 	);
+
 	if (!result.length) {
 		throw new Error(`Reconciliation order with id ${id} not found`);
 	}
+
 	return result[0];
 }
+export async function getReconciliationOrderLines(db: DB, id: number): Promise<ReconciliationOrderLine[]> {
+	const result = await db.execO<ReconciliationOrderLine>(
+		`SELECT * FROM reconciliation_order_lines
+		WHERE reconciliation_order_id = ?;`,
+		[id]
+	);
+
+	return result;
+}
 
-export async function createReconciliationOrder(db: DB, supplierOrderIds: number[]) {
+export async function createReconciliationOrder(db: DB, supplierOrderIds: number[]): Promise<number> {
 	if (!supplierOrderIds.length) {
 		throw new Error("Reconciliation order must be based on at least one supplier order");
 	}
 
-	const recondOrder = await db.execA(
+	const recondOrder = await db.execO<{ id: number }>(
 		`INSERT INTO reconciliation_order (supplier_order_ids) VALUES (json_array(${multiplyString(
 			"?",
 			supplierOrderIds.length
 		)})) RETURNING id;`,
 		supplierOrderIds
 	);
-	return recondOrder[0][0];
+	return recondOrder[0].id;
 }
 
 export async function addOrderLinesToReconciliationOrder(db: DB, id: number, isbns: string[]) {
-	const reconOrder = await db.execO<{ supplierOrderIds: number[] }>(
-		"SELECT supplier_order_ids as supplierOrderIds FROM reconciliation_order WHERE id = ?;",
-		[id]
-	);
+	const reconOrder = await db.execO<ReconciliationOrder>("SELECT * FROM reconciliation_order WHERE id = ?;", [id]);
 
 	if (!reconOrder[0]) {
 		throw new Error(`Reconciliation order ${id} not found`);
 	}
 
-	await db.exec(
-		`UPDATE reconciliation_order SET customer_order_line_ids = (json_array(${multiplyString("?", isbns.length)})) WHERE id = ?;`,
-		[...isbns, id]
-	);
+	const params = isbns.map((isbn) => [id, isbn]).flat();
```

Comment:
I think we should work with an array of `{isbn, quantity}[]` for the "new lines" parameter. We would need to update this map, remove the fixed `1` quantity below in `VALUES (...)` to reference the mapped quantity, and update `quantity + 1` to reference `excluded.quantity`. This gives us the foundation to take another pass at your debounce logic in the reconcilation view

Reply:
what about the reconciliation_order_id?